substrate/substrate-processor/src/mapping.ts

```diff
@@ -405,6 +405,10 @@ export function setUpItems(block: Block): void {
                 rec.parentCall = prev
                 populateSubcalls(prev, rec)
             }
+            if (prev.parentCall && isSubcall(prev.parentCall, rec)) {
```

Comment:
nested calls are handled incorrectly.

[1, 1, 1],  [1, 1], [1]  -> the first will be child of the second, the second will be cild of the third

[1, 1, 1],  [1]  -> the first will be child of the third

Reply:
yeah, i also expect that it works like that

Reply:
but i don't see the problem...

Reply:
ok if we talk about `[1, 1, 1], [1]` then there will be wrong parent assignment...
the parent will have the child as a subcall but i believe `[1]` shouldn't be in `.parentCall`

Reply:
idk that should be the correct behavior here, but `isSubcall` assumes that `[1, 1, 1]` is a subcall of `[1]`

Reply:
it is expected according to subcalls query - https://github.com/subsquid/archive.py/blob/master/sqa/substrate/query.py#L422-L426

Reply:
> it is expected according to subcalls query -
Bot sure, I guess it only says "request all subcalls and thier child", but nothing about how they should be grouped

Reply:
yeah, another way is to put only direct "children" into `subcalls` array. i guess both options are fine...
@eldargab how subcalls are supposed to work?

Reply:
i got confirmation that subcalls represent all children